packages/web-pkg/src/components/Modals/SaveAsModal.vue

```diff
@@ -0,0 +1,204 @@
+import { isShareSpaceResource, Resource, SpaceResource } from '@ownclouders/web-client'
+import { unref } from 'vue'
+import { resolveFileNameDuplicate } from '../../helpers'
+import { useGettext } from 'vue3-gettext'
+
+export default defineComponent({
+  name: 'SaveAsModal',
+  components: { AppLoadingSpinner },
+  props: {
+    modal: { type: Object as PropType<Modal>, required: true },
+    parentFolderLink: { type: Object as PropType<RouteLocationRaw>, required: true },
+    originalResource: { type: Object as PropType<Resource>, required: true },
+    content: { type: String, required: true }
+  },
+  setup(props) {
+    const iframeRef = ref<HTMLIFrameElement>()
+    const isLoading = ref(true)
+    const themeStore = useThemeStore()
+    const { $gettext } = useGettext()
+    const router = useRouter()
+    const clientService = useClientService()
+    const { removeModal } = useModals()
+    const { showMessage, showErrorMessage } = useMessages()
+    const { getMatchingSpace } = useGetMatchingSpace()
+    const { getEditorRouteOpts } = useFileActions()
+
+    const parentFolderRoute = router.resolve(props.parentFolderLink)
+    const iframeTitle = themeStore.currentTheme.common?.name
+    const iframeUrl = new URL(parentFolderRoute.href, window.location.origin)
+    iframeUrl.searchParams.append('hide-logo', 'true')
+    iframeUrl.searchParams.append('embed', 'true')
+    iframeUrl.searchParams.append('embed-target', 'location')
+    iframeUrl.searchParams.append('embed-choose-file-name', 'true')
+    iframeUrl.searchParams.append('embed-choose-file-name-suggestion', props.originalResource.name)
+
+    const onLoad = () => {
+      isLoading.value = false
+      unref(iframeRef).contentWindow.focus()
+    }
+
+    const onLocationPick = async ({ data }: MessageEvent) => {
+      if (data.name !== 'owncloud-embed:select') {
+        return
+      }
+
+      const { resources, fileName, locationQuery }: embedModeLocationPickMessageData = data.data
+
+      const destinationFolder: Resource = resources[0]
+      const space = getMatchingSpace(destinationFolder)
+
+      try {
+        const resource = await saveFile({ destinationFolder, fileName, space })
+        showMessage({
+          title: $gettext('"%{fileName}" was saved successfully', { fileName: resource.name })
+        })
+        openFile({ resource, space, locationQuery })
+      } catch (e) {
+        console.error(e)
+        showErrorMessage({
+          title: $gettext('Unable to save "%{fileName}"', { fileName }),
+          errors: [e]
+        })
+        console.error(e)
+      }
+
+      removeModal(props.modal.id)
+    }
+
+    const saveFile = async ({
+      destinationFolder,
+      fileName,
+      space
+    }: {
+      destinationFolder: Resource
+      fileName: string
+      space: SpaceResource
+    }) => {
+      const { children: existingResources } = await clientService.webdav.listFiles(space, {
+        fileId: destinationFolder.fileId
+      })
```

Comment:
You could limit the requested dav properties to reduce the request payload by quite a bit since you probably only need the name.